ui/packages/components/src/CodeBlock/CommandBlock.tsx

```diff
@@ -0,0 +1,180 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+import { CopyButton } from '@inngest/components/CopyButton';
+import { useCopyToClipboard } from '@inngest/components/hooks/useCopyToClipboard';
+import { cn } from '@inngest/components/utils/classNames';
+import { FONT, LINE_HEIGHT, createColors, createRules } from '@inngest/components/utils/monaco';
+import Editor, { useMonaco } from '@monaco-editor/react';
+import * as Tabs from '@radix-ui/react-tabs';
+import { type editor } from 'monaco-editor';
+
+import { isDark } from '../utils/theme';
+
+export type TabsProps = {
+  title: string;
+  content: string;
+  readOnly?: boolean;
+  language?: string;
+};
+
+type MonacoEditorType = editor.IStandaloneCodeEditor | null;
+
+const CommandBlock = ({ currentTabContent }: { currentTabContent?: TabsProps }) => {
+  const [dark, setDark] = useState(isDark());
+  const editorRef = useRef<MonacoEditorType>(null);
+  const wrapperRef = useRef<HTMLDivElement>(null);
+
+  const monaco = useMonaco();
+  const activeTabContent = currentTabContent || { content: '', readOnly: true, language: 'json' };
+
+  useEffect(() => {
+    // We don't have a DOM ref until we're rendered, so check for dark theme parent classes then
+    if (wrapperRef.current) {
```

Comment:
just curious, does this cause a light flicker before dark?

Reply:
This is the same logic you implemented in the CodeBlock. We need to change that theme logic, as besides the flicker when applying the theme, it also doesn't automatically change it to the other theme when moving between light and dark mode without a render, but I don't wanna mess with it in this PR